model/version/version.go

```diff
@@ -75,3 +76,54 @@ var (
 	BuildStatusActivateAtKey = bsonutil.MustHaveTag(BuildStatus{}, "ActivateAt")
 	BuildStatusBuildIdKey    = bsonutil.MustHaveTag(BuildStatus{}, "BuildId")
 )
+
+type DuplicateVersionsID struct {
+	Hash      string `bson:"hash"`
+	ProjectID string `bson:"project_id"`
+}
+
+type DuplicateVersions struct {
+	ID       DuplicateVersionsID `bson:"_id"`
+	Versions []Version           `bson:"versions"`
+}
+
+func FindDuplicateVersions(since time.Time) ([]DuplicateVersions, error) {
+	pipeline := []bson.M{
+		bson.M{
+			"$match": bson.M{
+				RequesterKey: evergreen.RepotrackerVersionRequester,
+				CreateTimeKey: bson.M{
+					"$gte": since,
+				},
+			},
+		},
+		bson.M{
+			"$group": bson.M{
+				"_id": bson.M{
+					"project_id": "$" + IdentifierKey,
+					"hash":       "$" + RevisionKey,
+				},
+				"count": bson.M{
+					"$sum": 1,
+				},
+				"versions": bson.M{
+					"$push": "$$ROOT",
+				},
+			},
+		},
+		bson.M{
+			"$match": bson.M{
+				"count": bson.M{
+					"$gt": 1,
+				},
+			},
+		},
+	}
+
+	out := []DuplicateVersions{}
+	if err := db.Aggregate(Collection, pipeline, &out); err != nil {
+		return nil, err
```

Comment:
I would probably wrap this, but I don't know if it matters. 

We're not super consistent about this, but it's also unclear if "no results found" is the same kind of error as "I can't talk to the database," and this doesn't quite expose that.

Reply:
In this case, Len of 0 means there are no duplicates, which isn't an error.

Reply:
There is a case where mgo returns a "ErrorNotFound", in this case that we'd be handling incorrectly.

Reply:
I tried running this on an empty collection, and mgo.ErrNotFound is not returned from db.Aggregate